runtime/lua/vim/lsp/handlers.lua

```diff
@@ -22,38 +22,38 @@ M[ms.workspace_executeCommand] = function(_, _, _, _)
 end
 
 --- @see # https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
----@param result lsp.ProgressParams
+---@param params lsp.ProgressParams
 ---@param ctx lsp.HandlerContext
-M[ms.dollar_progress] = function(_, result, ctx)
+M[ms.dollar_progress] = function(_, params, ctx)
```

Comment:
Most of the other handlers in this file name this parameter `result`. Should that be changed? Or is this PR specific to the Progress event? If `result` is generally the right name for this parameter, I suggest keeping this parameter name as `result`, and only change the `data` key.

Reply:
The handler here is for a notification (sent from server to the client), `params` is the right choice here. (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#notificationMessage)  
Most of the other handlers are response handlers. `result` is the right choice them.  (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#responseMessage)

That we put response, server->client request and notification handlers into the same table can be a bit confusing.

Reply:
Please see #27101: in short, all "request" and "notification" handlers (but not "response" handlers) should have `params` instead of `result`.